### PR TITLE
Fixed nuget version casing mismatch

### PR DESCRIPTION
**Pull request by davidfowl**
- Always update the nuspec with the resolved version and id on
install.

#2970

### CODE REVIEW

File: misc/MismatchedVersionCasing/src/A/project.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "frameworks": {
+    "dnx451": { }
+  },
+  "dependencies": {
+    "Contentful.NET": "0.0.5-Alpha"
```
Comment (moozzyk) [inline misc/MismatchedVersionCasing/src/A/project.json]: Does this test depend on this package being on NuGet?

Comment (davidfowl) [inline misc/MismatchedVersionCasing/src/A/project.json]: Yes, like all of our other tests and our source code :smile:

Comment (moozzyk) [inline misc/MismatchedVersionCasing/src/A/project.json]: I think other tests mostly depend on packages we kind of control, no?

File: src/Microsoft.Dnx.Tooling/Utils/NuGetPackageUtils.cs [hunk 2]
```diff
@@ -77,23 +77,21 @@ internal static class NuGetPackageUtils
                     // Fixup the casing of the nuspec on disk to match what we expect
                     var nuspecFile = Directory.EnumerateFiles(targetPath, "*" + NuGet.Constants.ManifestExtension).Single();
 
-                    if (!string.Equals(nuspecFile, targetNuspec, StringComparison.Ordinal))
```
Comment (moozzyk) [inline src/Microsoft.Dnx.Tooling/Utils/NuGetPackageUtils.cs]: Have you considered doing IgnoreCase check *and* printing the warning if the names still don't match? (I don't feel very strongly about about that though...)

Comment (davidfowl) [inline src/Microsoft.Dnx.Tooling/Utils/NuGetPackageUtils.cs]: Nah, doesn't matter. If we found it then we should use it.

File: misc/MismatchedVersionCasing/src/A/project.json (hunk 1 above)
Comment (davidfowl) [inline misc/MismatchedVersionCasing/src/A/project.json]: Some, some dont

**Pull request closed without merging**